modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/AtomicFields.scala

```diff
@@ -32,97 +32,98 @@ object AtomicFields {
-      AtomicField(name = "se_property",        _.se_property        ),
-      AtomicField(name = "tr_orderid",         _.tr_orderid         ),
-      AtomicField(name = "tr_affiliation",     _.tr_affiliation     ),
-      AtomicField(name = "tr_city",            _.tr_city            ),
-      AtomicField(name = "tr_state",           _.tr_state           ),
-      AtomicField(name = "tr_country",         _.tr_country         ),
-      AtomicField(name = "ti_orderid",         _.ti_orderid         ),
-      AtomicField(name = "ti_sku",             _.ti_sku             ),
-      AtomicField(name = "ti_name",            _.ti_name            ),
-      AtomicField(name = "ti_category",        _.ti_category        ),
-      AtomicField(name = "useragent",          _.useragent          ),
-      AtomicField(name = "br_name",            _.br_name            ),
-      AtomicField(name = "br_family",          _.br_family          ),
-      AtomicField(name = "br_version",         _.br_version         ),
-      AtomicField(name = "br_type",            _.br_type            ),
-      AtomicField(name = "br_renderengine",    _.br_renderengine    ),
-      AtomicField(name = "br_lang",            _.br_lang            ),
-      AtomicField(name = "br_colordepth",      _.br_colordepth      ),
-      AtomicField(name = "os_name",            _.os_name            ),
-      AtomicField(name = "os_family",          _.os_family          ),
-      AtomicField(name = "os_manufacturer",    _.os_manufacturer    ),
-      AtomicField(name = "os_timezone",        _.os_timezone        ),
-      AtomicField(name = "dvce_type",          _.dvce_type          ),
-      AtomicField(name = "doc_charset",        _.doc_charset        ),
-      AtomicField(name = "tr_currency",        _.tr_currency        ),
-      AtomicField(name = "ti_currency",        _.ti_currency        ),
-      AtomicField(name = "base_currency",      _.base_currency      ),
-      AtomicField(name = "geo_timezone",       _.geo_timezone       ),
-      AtomicField(name = "mkt_clickid",        _.mkt_clickid        ),
-      AtomicField(name = "mkt_network",        _.mkt_network        ),
-      AtomicField(name = "etl_tags",           _.etl_tags           ),
-      AtomicField(name = "refr_domain_userid", _.refr_domain_userid ),
-      AtomicField(name = "domain_sessionid",   _.domain_sessionid   ),
-      AtomicField(name = "event_vendor",       _.event_vendor       ),
-      AtomicField(name = "event_name",         _.event_name         ),
-      AtomicField(name = "event_format",       _.event_format       ),
-      AtomicField(name = "event_version",      _.event_version      ),
-      AtomicField(name = "event_fingerprint",  _.event_fingerprint  ),
+      AtomicField(name = "app_id", _.app_id, _.app_id = null),
+      AtomicField(name = "platform", _.platform, _.platform = null),
+      AtomicField(name = "event", _.event, _.event = null),
+      AtomicField(name = "event_id", _.event_id, _.event_id = null),
```

Comment:
For some loaders `event_id` is a a required field in the table.  For example, see [here](https://github.com/snowplow-incubator/snowflake-loader/blob/0.2.2/modules/core/src/main/scala/com.snowplowanalytics.snowplow.snowflake/processing/TableManager.scala#L147) how we create the snowflake events table.

Same is true of `v_collector` and `v_etl`, although I cannot tell you why it is this way!!

I'm fine with setting fields to null, and I understand how this helps with the wider design of incomplete events.  But be aware this makes it technically possible that enrich can emit events which cannot be loaded.

cc @colmsnowplow

Reply:
thanks for the heads up @istreeter ! That makes me think if we should
- keep a list of atomic field names to be excluded from this check or
- take prefix of the field as much as its' limit allow or
- let it be and accept the possibility